src/main/java/org/spongepowered/api/entity/living/player/CooldownTracker.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.entity.living.player;
+
+import org.spongepowered.api.item.ItemType;
+
+import java.util.OptionalDouble;
+import java.util.OptionalInt;
+
+/**
+ * Provides access to the item cooldowns of a {@link Player}.
+ */
+public interface CooldownTracker {
+
+    /**
+     * Checks if the specified {@link ItemType} is currently on cooldown
+     * for the player.
+     *
+     * @param type The item type to check is on cooldown
+     * @return Whether or not the specified item type is cooldown
+     */
+    default boolean hasCooldown(final ItemType type) {
+        return !getCooldown(type).isPresent();
```

Comment:
what? why `!`? this should return `true` if there *is* a cooldown

Reply:
You're quite correct, oops. I overrode this in the implementation so I never noticed.

Reply:
Just don't default implement this as every implementation is almost certainly going to have a more efficient implementation of it. Default implementing should only really be used for calling alternatives.